src/com/lilithsthrone/game/inventory/clothing/ClothingType.java

```diff
@@ -6358,13 +6359,35 @@ public String unequipText(GameCharacter clothingOwner, GameCharacter clothingRem
 	private static Map<String, String> oldIdConversionMap = new HashMap<>();
 	
 	public static AbstractClothingType getClothingTypeFromId(String id) {
+		return getClothingTypeFromId(id, null);
+	}
+	
+	public static AbstractClothingType getClothingTypeFromId(String id, String slotHint) {
 //		System.out.print("ID: "+id);
 		
 		if(oldIdConversionMap.containsKey(id)) {
 			id = oldIdConversionMap.get(id);
 		}
 		
-		id = Util.getClosestStringMatch(id, idToClothingMap.keySet());
+		Map<String, AbstractClothingType> choiceMap = idToClothingMap;
+		if (slotHint != null) {
```

Comment:
I've changed this in my local code to:
`if (slotHint!=null && !slotHint.isEmpty()) {`

Without the empty String check, my error log was being spammed with warnings about InventorySlot not having a valid value for an empty String.

Reply:
Interesting, I hadn't seen that myself, that I remember.